linux_new/Jenkinsfile

```diff
@@ -0,0 +1,887 @@
+        break
+    }
+} catch (Exception e) {
+    error "Error While Checking URL ${Target}: ${e.message}"
+}
+}
+// Function Definitions End
+
+/* Pipeline Declaration */
+pipeline {
+    agent none  // No Default Agent
+    options {
+        timeout(time: 2, unit: 'HOURS')
+    }
+
+    parameters {
+        string(name: 'TAG', defaultValue: 'jdk-23+37', description: 'Release Tag')
+        string(name: 'VERSION', defaultValue: 'jdk23', description: 'Release Version')
+        string(name: 'ARTIFACTS_TO_COPY', defaultValue: '**/alpine-linux/aarch64/temurin/*.tar.gz,**/alpine-linux/aarch64/temurin/*.zip,**/alpine-linux/aarch64/temurin/*.sha256.txt,**/alpine-linux/aarch64/temurin/*.msi,**/alpine-linux/aarch64/temurin/*.pkg,**/alpine-linux/aarch64/temurin/*.json,**/alpine-linux/aarch64/temurin/*.sig', description: 'Artifacts String')
+        booleanParam(name: 'RELEASE', defaultValue: false, description: 'Release Flag' )
+        booleanParam(name: 'DRY_RUN', defaultValue: false, description: 'Release Dry Run')
+        booleanParam(name: 'ENABLEGPGSIGNING', defaultValue: true, description: 'Require GPG Signing')
+        booleanParam(name: 'ENABLEDEBUG', defaultValue: false, description: 'Tick to enable --stacktrace for gradle build')
+        booleanParam(name: 'REPACKAGE', defaultValue: false, description: 'Tick if this is a republish of an existing package, ie xx.xxx.2 , rather than the base release of a package (1)')
+        string(name: 'PACKAGE_INCREMENT', defaultValue: '1', description: 'This is the incremental number used for re-releases of package versions - Should be set appropriately if RePackage is True')
+        string(name: 'TEMURIN_VERSION_INCREMENT', defaultValue: '0', description: 'Final Element Of The Version Number - Used For Temurin Specific Patches - NEARLY ALWAYS ZERO')
+    }
+    // Stage Definition - Start
+    stages {
+        // Print Parameters Stage - Start
+        stage('Print Parameters') {
+            agent { label NODE_LABEL }
+            steps {
+                script {
+                    echo "Entering Stage : Print Parameters"
+                    echo "Tag : ${params.TAG}"
+                    echo "Version : ${params.VERSION}"
+                    echo "Artifacts : ${params.ARTIFACTS_TO_COPY}"
+                    echo "Release : ${params.RELEASE}"
+                    echo "Dry Run : ${params.DRY_RUN}"
+
+                    if (params.DRY_RUN) {
+                        echo "This Is A Dry Run - And As No Packages Are Potentially On Github - No Further Stages Will Run"
+                    } else {
+                        echo "Not A Dry Run - About To Build And Publish Packages"
+                    }
+                }
+            }
+        }
+        // Print Parameters Stage - End
+        // Process Parameters Stage - Start
+        stage('Process Parameters') {
+            agent { label NODE_LABEL }
+            when {
+                expression { return params.DRY_RUN == false }
```

Comment:
My brain struggles to parse this but I expect it's been tested and will skip it if it's a `DRY_RUN`. I wonder if it's possible to make it a more readable expression (or just stick a comment above it)?

Reply:
Its driven by the parent job, so inherits both the parameter name, and its state.. when I rework some of this, I'll see if I can do something nicer on the brain.